src/legacy/ui/public/vis/vis_types/_vislib_vis_legend.scss

```diff
@@ -81,32 +91,55 @@
 }
 
 .visLegend__valueTitle {
+  @include euiTextTruncate; // ALWAYS truncate
   color: $visTextColor;
 
   &:hover {
     text-decoration: underline;
   }
 }
 
-.visLegend__valueTitle--truncate {
-  @include euiTextTruncate;
-}
-
-.visLegend__valueTitle--full {
-  word-break: break-all;
+.visLegend__valueTitle--full ~ .visLegend__valueDetails {
+  z-index: 2;
 }
 
 .visLegend__valueDetails {
-  border-bottom: 1px solid $euiColorLightShade;
-  padding-bottom: $euiSizeXS;
+  background-color: $euiColorEmptyShade;
+
+  .visLib--legend-left &,
+  .visLib--legend-right & {
+    margin-top: $euiSizeXS;
+    border-bottom: $euiBorderThin;
+  }
+
+  .visLib--legend-top &,
+  .visLib--legend-bottom & {
+    @include euiBottomShadowMedium;
+    position: absolute;
+    border-radius: $euiBorderRadius;
+  }
+
+  .visLib--legend-bottom & {
+    bottom: $visLegendLineHeight + $euiSizeXS;
+    margin-bottom: $euiSizeXS;
```

Comment:
To clean these two lines you could just add that margin bottom amount to the `bottom` so it's: 
```
bottom: $visLegendLineHeight + $euiSizeS;
```